.github/workflows/cherry-pick.yml

```diff
@@ -1,26 +1,328 @@
 name: Cherry Pick Tool
 on:
+  issues:
+    types: [milestoned, labeled]
   pull_request:
-    types: [closed, edited]
+    types: [closed]
   workflow_dispatch:
     inputs:
       release_branch:
         description: Provide the release branch you want to cherry pick into. Example release/6.9
         default: ''
         required: true
       pull_requests:
-        description: The pull request number. Separate pull request numbers with a comma (no space) if more than one.
+        description: The pull request number.
+        default: ''
         required: true
+      skipSlackPing:
+        description: "Skip Slack Ping: If true, the Slack ping will be skipped (useful for testing)"
+        type: boolean
+        required: false
+        default: false
+      slackChannelOverride:
+        description: "Slack Channel Override: The channel ID to send the Slack ping about the code freeze violation"
+        required: false
+        default: ''
 
 concurrency: 
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+    GIT_COMMITTER_NAME: 'WooCommerce Bot'
+    GIT_COMMITTER_EMAIL: 'no-reply@woocommerce.com'
+    GIT_AUTHOR_NAME: 'WooCommerce Bot'
+    GIT_AUTHOR_EMAIL: 'no-reply@woocommerce.com'
+
 jobs:
+  verify:
+    name: Verify
+    runs-on: ubuntu-20.04
+    outputs:
+      run: ${{ steps.check.outputs.run }}
+    steps:
+      - name: check
+        id: check
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let run = false;
+
+            if ( context.payload.inputs && context.payload.inputs.release_branch && context.payload.inputs.release_branch != null ) {
+              run = true;
+            }
+
+            if ( context.payload.issue && context.payload.issue.pull_request != null && context.payload.issue.pull_request.merged_at != null && context.payload.issue.milestone != null  ) {
+              run = true;
+            }
+
+            if ( context.payload.pull_request && context.payload.pull_request != null && context.payload.pull_request.merged == true && context.payload.pull_request.milestone != null ) {
+              run = true;
+            }
+
+            if ( context.payload.pull_request && ( context.payload.pull_request.user.login == 'github-actions[bot]' || context.payload.pull_request.user.type == 'Bot' ) ) {
+              run = false;
+            }
```

Comment:
Sorry, I meant to leave this with my review, but somehow GitHub swallowed by drafted comment.

I found this section of logic a bit difficult to follow. This pattern is usually used when you're looping over something and exiting early, or something like that. Could we name the conditionals `is[SomeDescriptiveName]`, and then compare them at the end?

Something like `const run = ! isBotPR  && ( isProvidedByInput || isMilestonedIssue || isMilestonedPR )`

Reply:
So I did this because doing it inline got too long and even more confusing to read. But open to other suggestions.

Reply:
@jonathansadowski so I pushed up some updates to that logic. Let me know if that is sort of what you mean?

Reply:
Yep, thanks... that's what I was thinking. The part I was most worried about is that another condition would be added after the last `run=false` part, and it'd potentially break the existing logic. The logic is much more explicit this way.